cmd/build/v2/build.go

```diff
@@ -320,8 +325,15 @@ func (bc *OktetoBuilder) buildSvcFromDockerfile(ctx context.Context, manifest *m
 	bc.ioCtrl.Logger().Info(fmt.Sprintf("Building service '%s' from Dockerfile", svcName))
 	isStackManifest := manifest.Type == model.StackType
 	buildSvcInfo := bc.getBuildInfoWithoutVolumeMounts(manifest.Build[svcName], isStackManifest)
-	buildHash := bc.hasher.hashService(buildSvcInfo)
-	tagToBuild := newImageTagger(bc.Config).getServiceImageReference(manifest.Name, svcName, buildSvcInfo, buildHash)
+	var err error
+	var buildHash string
```

Comment:
if smart build is not enabled `buildHash` is empty. It is expected?

Reply:
Yes, We were already checking checking in the tagger if the hash is empty to create it with the hash or using the okteto tag. Am I missing something?

Reply:
I find it a bit confusing. I think the best thing to do would be at the time of checking if it is enabled or not to put the appropriate tag if possible. This way it is difficult to know at this point that the tagger has logic to control it.

Reply:
But I think that if we are not calculating the buildHash it should be empty. Otherwise the tagger should have logic for the smart build or viceversa